---
title: "Final Report: Impact of diet on diversity and abundance of gut microbiota"
author: "Karan Lala"
date: "December 6, 2018"
output:
    github_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

###E-mail: "krlala@usfca.edu"

# Introduction


It seems intuitive that gut microbiota depends on the sort of diet provided
as the processing of such material would likely involve some metabolic processes
on the part of the members of the gut microbiota. The gut microbiota plays a
major role in the development of food absorption and low  grade inflammation,
two key processes in obesity and diabetes.[@david2014diet] 

A majority of the gut microbiota
reside in the large intestine and they can even increase energy production from
diets. The gut microbiota hence can have an immense impact of the health and
physiological well being of individuals.[@rook2005microbes] Apart from a lack of clear understanding
as to how the gut microbiota affect obese individuals, there aren't many studies
that have explored its connection to conditions like anorexia nervosa [@armougom2009monitoring]. A gut-brain
axis was identified in recent papers, and composition or change in gut microbiota
is linked to changes in moods and bouts of depression.[@foster2013gut]
The microbiota produces an array of bioactive metabolic products capable of
entering systemic circulation. It plays a part in immune modulation, metabolic
activities, digestion and differs across humans. Humans, like all other animals
have genetically adapted to their surroundings and diet, which in turn influence
the gut microbiota. Just as different soils have a different composition of
bacteria, and how specific species of bacteria are differentially active in
environments that differ in any number of ways, (like moisture, for example)
gut microbiota also present in very different compositions.[@fierer2017embracing][@turnbaugh2009core]
Studies have revealed that antibiotic treatment during the first months of
life is associated with being overweight later in life. This could happen due to
the reduction in diversity caused by antibiotics, and also interference with
beneficial effects of breast milk.[@ajslev2011childhood][@rautava2016early] Composition of the gut microbiota is sensitive
during the early years of life and plays a key role in human health and establishing
a healthy microbiota. 


Recent evidence suggests that a failure to enhance
microbial communities during early life can result in cases of obesity later.[@gill2006metagenomic]
The microbiome also metabolizes plant polysaccharides (such as fiber and resistant starch)
into short-chain fatty acids, like butyrate is an energy source for intestinal epithelial
cells and may protect against carcinogenic changes in these cells; propionate acts
as a signaling molecule in the colon wall; and acetate modulates gene expression
in the gastrointestinal tract and also in other tissues.[@koliada2017association]
In certain studies in mice, exposure to antibiotic doses resulted in a significant
decrease in the B/F ratio, which indicates lower diversity and a propensity towards
obesity.[@ley2006microbial]
As a whole, there isn't much evidence suggesting that there exists a core human
gut microbiome. While there are similarities between communities and families,
each human has a unique microbiome, and across populations and changes in climate
and diet, differences in the gut microbiota cannot be ruled out.[@yatsunenko2012human]

The Neolithic revolution brought with it agriculture and animal husbandry
roughly 10,000 years ago.[@mira2006neolithic] That also means, that then began a period with more
stable and abundant food resources. It is suggested that Human microbiota changed
significantly because of such changes, and the hypothesis may have some merit.
Gut microbiota are known to vary with diversity in carnivores, omnivores and
herbivores due to their difference in diet and phylogeny.
Westernized countries have observed an increase in autoimmune disorders, due to
increased hygiene and change in gut microbiota.[@cordain2005origins] The dataset used in the study,
"Impact of diet in shaping gut microbiota revealed by a comparative study in
children from Europe and rural Africa" looks at the gut microbiota of young
children upto 6 years old with two different environments and diets.

Most of the sequences found in the samples are from the four most populated
bacterial phyla which are Actinobacteria, Bacteroidetes, Firmicutes, and
Proteobacteria. [@ventura2007genomics]There are other studies that have also found the mentioned four
phyla as comprising a majority of the human gut microbiome.
However, across these phyla, there are differences found in both groups of samples.
The microbiota from the BF children were found to have a higher percentage
of Actinobacteria and Bactroidetes than the EU children, who showed a higher
percentage of Firmicutes and Proteobacteria. While Actinobacteria and Bacteroidetes
were present in smaller amounts comparatively, Firmicutes and Proteobacteria
were significantly enriched in the EU children.[@mariat2009firmicutes]
Prevotella, Xylanibacter from Bacteroidetes  and Treponema from Spirochaetes are
found only in the microbiota of BF children.


It is also noted that the BF children that are youngest and breast fed observe
abundant Actinobacteria from the Bifidobacterium genus. This supports the hypothesis
that diet affects the gut microbiota. After hierarchical clustering, it is observed
that the microbiome of the youngest children form a cluster, separate from the
clusters formed by the other subjects, representing the EU and BF children.
There is a higher microbial richness observed in BF samples, possibly due to
more exposure to a number of environmental microbes that populate the food or
the high-fiber diet followed by the BF children. Western diets rich with sugars,
fat and protein could result in a less enriched microbiome, and much lower
numbers of non infectious colonic diseases are observed in Africans following
a high fiber diet.


Diet plays an important role in determining the composition of the microbiota. A
study helped show that as bacterial species that are associated with high sugar
and fat diets are found to promote obesity in gnotobiotic mice.[@turnbaugh2009effect] The ratio of
Firmicutes to Bacteroidetes (F/B) differs in obese and lean humans and the
ratio decreases on a low calorie diet with weight loss.


# Methods

## Study design

The study characterizes the fecal microbiota of 14 healthy children from the
Mossi ethnic group, living in a small village of Boulpon in Burkina Faso.
This is contrasted with samples from 15 healthy children living in an urban
environment in Florence, Italy. The children from BF represented consumers of
a traditional rural African diet which is low in fat and animal protein but rich
in starch, fiber and plant polysaccharides. They have a high content of cereals,
fibers and sparse amount of chicken and termites. Children are breast fed upto
the age of two years.
The EU children are breast fed till the age of one year, to represent a Western
population and diet. Typical diet includes a high amount of animal protein, sugar,
starch, and low in fiber. The number of calories differ as well, with the BF
children from the ages of 1 to 2 consuming 672.2kcal/d, and 2-6 year olds
consuming 996 kcal/d. The calories consumed for the EU children for ages 1-2 and
2-6 are 1068.7 kcal/d and 1512.7 kcal/d respectively.
That amounts to between a 50-60 percent increase in calories across both age
groups for the EU children.

## Sample origin and sequencing


Fecal samples were collected by physicians and preserved in RNAlater (Qiagen)
at ???80 °C until extraction of genomic DNA. The 16S rRNA genes for each sample
was amplified using a primer set specific for V5 and V6 hypervariable 16S region.
The forward primer was attached with a barcode sequence and pyrosequencing was
carried out using a 454 Life Sciences Genome Sequencer FLX instrument.[@rothberg2008development]
The samples were obtained from children who had not taken any antibiotics or
probiotics in the 6 months before the sampling date and all individuals seemed
healthy.

## Computational

DADA2 starts with set of Illumina-sequenced paired-end fastq files that have been split, sample wise and from which the barcodes/adapters have already
been removed.[@callahan2016dada2] The final product is an amplicon sequence variant (ASV) table, a higher-resolution analogue of the traditional OTU table,
which records the number of times each exact amplicon sequence variant was observed in each sample.
Taxonomy is then assigned and data can be imported into the popular phyloseq R package for the analysis of microbiome data.


Quality profiles for forward and reverse reads are generated and DADA2 incorporates quality information into its models, for example accounting to Some
extent for the low base quality reads at the end of sequences. It then filters and trims the data. DADA2 builds an error model from the data, and starts
with an initial guess, as do machine learning techniques.
Dereplication combines all identical sequencing reads into into "unique sequences" with a corresponding "abundance" equal to the number of reads with
that unique sequence. Dereplication substantially reduces computation time by eliminating redundant comparisons. Now, DADA2 merges the forward and
reverse reads together to obtain the full denoised sequences and constructs a sequence variant table by traditional methods.
Next, it removes chimeras and provides a native implementation of the Bayesian classifier method for assigning taxonomy. 

Finally, this data generated can be used with phyloseq.
The phyloseq package is a tool to import, store, analyze, and graphically display complex phylogenetic sequencing data that has already been clustered into Operational Taxonomic Units (OTUs), especially when there is associated sample data, phylogenetic tree, and/or taxonomic assignment of the OTUs.[@mcmurdie2013phyloseq] This package leverages many of the tools available in R for ecology and phylogenetic analysis (vegan, ade4, ape, picante), while also using advanced/flexible graphic systems (ggplot2) to easily produce publication-quality graphics of complex phylogenetic data.


# Results



```{r your-code-here, include=FALSE}
# Add code chunks as needed for your analyses
# For most analyses, I would recommend splitting the
# intensive computational part into a seperate R script file
# and then just load your libraries and the data object here
# using the `load()` function. Ask me for clarification if this is
# unclear.
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")
library("citr")
library("phyloseq")
```

```{r load-packages-and-data, include=FALSE}
# load packages as needed
library("dplyr")
load("output/phyloseq_obj.Rdata")
load("output/melted_phyloseq.Rdata")
# load data produced from analysis scripts using
# something like load("output/processed_data.Rdata")
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")
```





```{r figure 1 phyluminEU, echo = FALSE}
# Subsets only EU samples
euphyloseq_obj <- subset_samples(phyloseq_obj,
                                     IHMC_ethnicity == "caucasian")
# plots abundance according to phyla
plot_bar(euphyloseq_obj,
         x = "Phylum")
```

 *Figure 1:* The bar plot in this figure shows the abundance of bacteria by phylum, for the children from the EU. The most abundant is Firmicutes, and Bacteroidetes is much lower. Proteobacteria and Actinobacteria make up the four major abundant phylum.
 
 
 
 
 
 
```{r figure 2 phyluminBF, echo = FALSE}
# subests only BF samples
bfphyloseq_obj <- subset_samples(phyloseq_obj,
                                     IHMC_ethnicity == "black african")
plot_bar(bfphyloseq_obj,
         x = "Phylum")
```

 *Figure 2:* This bar plot clearly shows the abundance of bacteria by phylum, for the BF children. Bacteroidetes is the most abundant, followd by Firmicutes and Actinobacteria. Proteobacteria and Spirochaetes are also observed. Even the NA (unidentified) phyla has some presence.
 
 
 
 
 
 
```{r figure 3 alphadiv_birthdelivery, echo = FALSE}
# Creates an alpha diversity plot.
# This uses the Shannon and Simpson method of
# measuring alpha diversity.
plot_richness(euphyloseq_obj,
              x = "birth_delivery",
              measures = c("Shannon", "Simpson"), color = "age"
              ) +
  xlab("Childbirth") +
  geom_jitter(width = 0.2) +
  theme_bw()
```

 *Figure 3:* Alpha diversity plot by Shannon and Simpson method, examining different birthing conditions. There aren't many samples in the cesarean childbirth. The markers are shaded for age, with the darker color representing a younger age.
 
 
 
 
 
 
```{r figure 4 alphadiv_sex, echo = FALSE}
# Another alpha diversity plot, but created with distinction
# according to sex.
plot_richness(phyloseq_obj,
              x = "Sex",
              measures = c("Shannon", "Simpson"),
              color = "IHMC_ethnicity") +
  xlab("Sex") +
  geom_jitter(width = 0.2) +
  theme_bw()
```

 *Figure 4:* Alpha diversity by Shannon and Simpson method, with sex on the x axis and the points shaded by ethnicity.
 There doesn't appear to be much difference in diversity by gender. For the ethnicity, it appears similar but with more data from the BF samples on the lower end of the index.
 
 
 
 
 
 
```{r figure 5 networkplot, echo = FALSE}
# creates an igraph using Bray-Curtis method for determining
# community similarity. max.dist is usually 0.3, changed to
# 1.0 to accomodate all points(samples)
ig <- make_network(phyloseq_obj, dist.fun = "bray", max.dist = 1.0)
plot_network(ig, phyloseq_obj,  color = "age", shape = "IHMC_ethnicity",
             line_weight = 0.4, label = "BioSample")
```         

 *Figure 5:* This network plot shows the similarity between gut microbiota from different samples, using the Bray-Curtis method. Since the shapes represent ethnicity, it is notable that all clusters are of the same ethnicity except for one on the top right of the image.
 
 
 
 
 
 
```{r figure 6 netplot, echo = FALSE}
plot_net(phyloseq_obj, maxdist = 1.0, point_label = "BioSample",
         color = "IHMC_ethnicity")
```         

 *Figure 6:* This is a network plot using the Jaccard distance index, and shows same ethnicity clusters apart for one as seen by the dissimilar colors representing ethnicity in the same cluster.
 
 




```{r figure 7 Order_insample, echo = FALSE}
# facet_grid allows for creation of multiple graphs
# separated according to parameter provided.
p <- plot_bar(phyloseq_obj, x = "IHMC_ethnicity", fill = "Order",
             facet_grid = ~BioSample)
# the following line of code fills the black separation lines
# between the different orders
p + geom_bar(aes(color = Order, fill = Order),
             stat = "identity", position = "stack")

```         

 *Figure 7:* This bar plot shows the abundance of bacteria, colored by their Order, for each sample. 






```{r figure 8 familyabundance, echo = FALSE}

plot_bar(phyloseq_obj, facet_grid = ~IHMC_ethnicity,
         x = "Family")
```

 *Figure 8:* This graph looks at the abundance by Family across both ethnicities.








# Discussion

Add around 3-4 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited
